core/src/main/java/edu/wpi/grip/core/events/SocketChangedEvent.java

```diff
@@ -31,6 +31,18 @@ public boolean isRegarding(Socket socket) {
     return socket.equals(this.socket);
   }
 
+  /**
+   * This event will only dirty the save if the InputSocket does not have connections.
+   * Thus the value can only have been changed by a UI component.
+   * If the socket has connections then the value change is triggered by another socket's change.
+   *
+   * @return True if this should dirty the save.
+   */
+  @Override
+  public boolean doesDirtySave() {
+    return socket.getDirection() == Socket.Direction.INPUT && socket.getConnections().isEmpty();
+  }
```

Comment:
This is the same code as the method below it.
Can you refactor it into one method that is called by both methods?

Or just have one of these methods call the other one.